packages/backend/src/managers/application/applicationManager.ts

```diff
@@ -176,45 +179,52 @@ export class ApplicationManager extends Publisher<ApplicationState[]> implements
   private async initApplication(
     connection: ContainerProviderConnection,
     recipe: Recipe,
-    model: ModelInfo,
+    model?: ModelInfo,
     labels: Record<string, string> = {},
   ): Promise<PodInfo> {
+    const modelId = model ? model.id : '0';
```

Comment:
I don't really like that

Reply:
Sure, me neither:
> Note: I'm not very happy with the way the applicationManager functions accept an optional model, depending on if we want to use llama-stack or not. Let's first discuss the high-level architecture of this solution, I can change these implementation details when we agree on the architecture

Reply:
Sorry I went to code too fast :pray:  

> Let's first discuss the high-level architecture of this solution,

What depends on the `modelId`? We use it for operation like `hasApplicationPod` & `removeApplication`. Maybe we should only use the recipeId as key?

Reply:
I'm not very comfortable changing the labels, as the model label is also used for the tasks (the user can start the same recipe with different models, and we want to differentiate the tasks for them).

I have pushed a new commit changing how the parameters are passed: having explicit options inferModel / startLlamaStack, which enforce the other parameters (the `model` parameter for the moment). I also changed this arbitrary `0` model Id to `<none>`